### PR TITLE
Xpress speed improvements

### PR DESCRIPTION
**Pull request by merraksh**
The main change is improved performance for problems with many equality AND many inequality constraints. Previously Xpress added both classes of constraints as one, causing performance loss due to a Python loop. Now it only uses API functions. Other additions: statistics on CPU time and objective value; an rST entry.

### CODE REVIEW

File: cvxpy/reductions/solvers/qp_solvers/xpress_qpif.py [hunk 1]
```diff
@@ -221,7 +210,7 @@ def solve_via_data(self, data, warm_start, verbose, solver_opts, solver_cache=No
         # Set options if compatible with Xpress problem control names
 
         self.prob_.setControl({i: solver_opts[i] for i in solver_opts
-                               if hasattr(xp.controls.__dict__, i)})
```
Comment (SteveDiamond) [inline cvxpy/reductions/solvers/qp_solvers/xpress_qpif.py:224]: Are you sure we don't need the ``hasattr``? Why was it there before?

**Review by SteveDiamond — APPROVED** [review]
It all looks good to me!

**Comment by SteveDiamond** [comment]
Thanks for the PR!

**Comment by merraksh** [comment]
Thank you!

**Pull request merged**